src/python/pants/backend/python/rules/targets.py

```diff
@@ -0,0 +1,207 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pathlib import PurePath
+from typing import Any, ClassVar, Optional
+
+from pants.build_graph.address import Address
+from pants.engine.objects import union
+from pants.engine.target import (
+    COMMON_TARGET_FIELDS,
+    BoolField,
+    PrimitiveField,
+    Sources,
+    SourcesResult,
+    StringField,
+    StringOrStringListField,
+    Target,
+)
+
+
+@union
+class PythonSources(Sources):
+    @classmethod
```

Comment:
MyPy didn't like when this was a static method because it complained about the line `super().validate_result`.

Reply:
It is correct to do so. I think it would fail at runtime because `super()` is basically sugar for `super(__class__, <firstarg>)`, and there is no `<firstarg>` here.